File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    time = 1 # 작업하는 날짜 수
    cnt = 0 # 작업이 종료되는 갯수
    
    while len(progresses) > 0 : # 작업의 갯수가 0일 때 종료
        
        # 작업 진도 + 속도 * 날짜 수 >= 100
        if progresses[0] + speeds[0]*time >= 100 :
            progresses.pop(0) # 작업이 완료된 것이므로 제거
            speeds.pop(0)
            cnt += 1 # 작업 종료 갯수 += 1
        
        # 100이 안넘을 경우
        else :
            if cnt >= 1 : # 종료된 갯수가 1개 이상이면
                answer.append(cnt) # 답 추가
                cnt = 0 # 종료 갯수 초기화
            
            time += 1 # 날짜 += 1
    
    answer.append(cnt) # 작업의 갯수가 0이 되었을 때 마지막 완료 갯수 추가
    
    return answer

File: 프로그래머스/lv2/42586. 기능개발/기능개발.kt
import java.util.*

class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        var answer = intArrayOf()
        var cnt = 0 // 한번에 배포되는 기능의 수
        var time = 1 // 작업이 완료되는 날짜
        
        var progressesQueue : Queue<Int> = LinkedList<Int>()
        var speedsQueue : Queue<Int> = LinkedList<Int>()
        
        // progresses, speeds를 Queue로 변환
        for(i in 0 until progresses.size){
            progressesQueue.add(progresses[i])
            speedsQueue.add(speeds[i])
        }
        
        // queue가 비어있지 않은 동안 반복
        while(!progressesQueue.isEmpty()){
            
            // 작업 진도 + (작업 속도 * 날짜) >= 100
            if(progressesQueue.element() + speedsQueue.element()*time >= 100){
                
                // 작업 제거 및 배포 기능 수 ++
                progressesQueue.remove()
                speedsQueue.remove()
                cnt++
            }
            
            // 100이 아니라면
            else {
                if(cnt >= 1){ // 완료된 작업 수가 1 이상이면
                    answer = answer.plus(cnt) // answer에 추가하고 작업 수 초기화
                    cnt = 0
                }
                
                time ++ // 날짜++
            }
        }
        answer = answer.plus(cnt) // 마지막 완료되는 작업 수 추가
        
        return answer
    }
}